app/controllers/application_controller.rb

```diff
@@ -5,13 +5,20 @@ class ApplicationController < ActionController::Base
     # Pundit: white-list approach.
   after_action :verify_authorized, except: :index, unless: :skip_pundit?
   after_action :verify_policy_scoped, only: :index, unless: :skip_pundit?
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+  def user_not_authorized
+    flash[:alert] = "You are not authorized to perform this action."
+    redirect_to(root_path)
+  end
```

Comment:
I don't think we're ready yet to go without the rails clues as to what has gone wrong.